File: python/tests/model/current_geo_location_test.py
import pytest

from openral_py.model.container import Container
from openral_py.model.current_geo_location import CurrentGeoLocation


class TestCurrentGeoLocation:

    def test_current_geo_location_from_map(self):
        # arrange
        data = {'container': {'UID': 'uid'}}
        
        # act
        result = CurrentGeoLocation.from_map(data)
        
        # assert
        assert isinstance(result, CurrentGeoLocation)
        assert isinstance(result.container, Container)
        assert result.container.uid == 'uid'

    def test_current_geo_location_from_map_with_empty_data(self):
        # arrange
        data = {}
        
        # act
        result = CurrentGeoLocation.from_map(data)
        
        # assert
        assert isinstance(result, CurrentGeoLocation)
        assert result.container is None

File: dart/test/model/current_geo_location_test.dart
import 'package:openral_core/src/model/container.dart';
import 'package:openral_core/src/model/current_geo_location.dart';
import 'package:openral_core/src/model/geo_coordinates.dart';
import 'package:openral_core/src/model/postal_address.dart';
import 'package:test/test.dart';

void main() {
  group(
    CurrentGeoLocation,
    () {
      group('toMap', () {
        test('returns map with values', () async {
          //arrange
          final currentGeoLocation = CurrentGeoLocation(
            container: Container(
              uid: "uid1",
            ),
            postalAddress: PostalAddress(
              cityName: "Hannover",
              cityNumber: "30657",
              streetName: "Example Street",
              streetNumber: "123a",
              country: "Germany",
            ),
            plusCode: "plusCode",
            threeWordCode: "threeWordCode",
            geoCoordinates: GeoCoordinates(
              longitude: 12.0,
              latitude: 21.0,
            ),
          );

          //act
          final map = currentGeoLocation.toMap();

          //assert
          expect(
            map,
            equals({
              "container": {
                "UID": "uid1",
              },
              "postalAddress": {
                "streetName": "Example Street",
                "streetNumber": "123a",
                "cityName": "Hannover",
                "cityNumber": "30657",
                "country": "Germany",
              },
              "3WordCode": "threeWordCode",
              "geoCoordinates": {
                "longitude": 12.0,
                "latitude": 21.0,
              },
              "plusCode": "plusCode",
            }),
          );
        });
      });
    },
  );
}
